### PR TITLE
DataTable - relocate columns

### PR DESCRIPTION
**Pull request by cristineguadelupe**
https://github.com/livebook-dev/kino/assets/5660941/5b07f737-a6d3-4272-8568-0dd8e32ac79e

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

File: lib/kino/table.ex [hunk 1]
```diff
@@ -8,14 +8,15 @@ defmodule Kino.Table do
...
+          :features => list(:export | :refetch | :pagination | :sorting | :relocate),
           optional(:export) => %{formats: list(String.t())}
         }
 
   @type rows_spec :: %{
           offset: non_neg_integer(),
           limit: pos_integer(),
-          order: nil | %{direction: :asc | :desc, key: term()}
+          order: nil | %{direction: :asc | :desc, key: term()},
+          relocate: list(%{start_index: non_neg_integer(), end_index: non_neg_integer()})
```
Comment (jonatanklosko) [inline lib/kino/table.ex:19]:
```suggestion
          relocates: list(%{from_index: non_neg_integer(), to_index: non_neg_integer()})
```

File: lib/kino/table.ex [hunk 2]
```diff
@@ -95,7 +96,8 @@ defmodule Kino.Table do
        # Data specification
        page: 1,
        limit: @limit,
-       order: nil
+       order: nil,
+       relocate: []
```
Comment (jonatanklosko) [inline lib/kino/table.ex:100]:
```suggestion
       relocates: []
```

File: lib/kino/table.ex [hunk 3]
```diff
@@ -154,6 +156,11 @@ defmodule Kino.Table do
     {:noreply, broadcast_update(ctx)}
   end
 
+  def handle_event("relocate", %{"start_index" => start_index, "end_index" => end_index}, ctx) do
+    relocate = ctx.assigns.relocate ++ [%{start_index: start_index, end_index: end_index}]
+    {:noreply, ctx |> assign(relocate: relocate) |> broadcast_update()}
+  end
```
Comment (jonatanklosko) [inline lib/kino/table.ex:162]:
```suggestion
  def handle_event("relocate", %{"from_index" => from_index, "to_index" => to_index}, ctx) do
    relocates = ctx.assigns.relocates ++ [%{from_index: from_index, to_index: to_index}]
    {:noreply, ctx |> assign(relocates: relocates) |> broadcast_update()}
  end
```

File: lib/kino/table.ex [hunk 4]
```diff
@@ -200,7 +207,8 @@ defmodule Kino.Table do
       max_page: total_rows && ceil(total_rows / ctx.assigns.limit),
       total_rows: total_rows,
       order: order,
-      limit: ctx.assigns.limit
+      limit: ctx.assigns.limit,
+      relocate: ctx.assigns.relocate
```
Comment (jonatanklosko) [inline lib/kino/table.ex:211]:
```suggestion
      relocates: ctx.assigns.relocates
```

File: lib/kino/table.ex [hunk 5]
```diff
@@ -254,7 +262,8 @@ defmodule Kino.Table do
     %{
       offset: (ctx.assigns.page - 1) * ctx.assigns.limit,
       limit: ctx.assigns.limit,
-      order: ctx.assigns.order
+      order: ctx.assigns.order,
+      relocate: ctx.assigns.relocate
```
Comment (jonatanklosko) [inline lib/kino/table.ex:266]:
```suggestion
      relocates: ctx.assigns.relocates
```

File: assets/data_table/src/App.js [hunk 6]
```diff
@@ -362,17 +367,29 @@ export function App({ ctx, data }) {
     });
   }, []);
 
-  const onHeaderMenuClick = useCallback((column, bounds) => {
-    const { summary, id, type } = columns[column];
-    if (!summary) {
-      setMenu({ column, bounds, columnKey: id, columnType: type });
-    }
+  const onColumnMoved = useCallback((startIndex, endIndex) => {
+    ctx.pushEvent("relocate", { start_index: startIndex, end_index: endIndex });
```
Comment (jonatanklosko) [inline assets/data_table/src/App.js:371]:
```suggestion
    ctx.pushEvent("relocate", { from_index: startIndex, to_index: endIndex });
```

**Review by jonatanklosko — APPROVED** [review]

**Comment by jonatanklosko** [comment]
🐈🍵

**Pull request merged**